src/zenml/cli/stack_recipes.py

```diff
@@ -0,0 +1,806 @@
+
+
+@stack_recipe.command(
+    help="Run the stack_recipe that you previously pulled with "
+    "`zenml stack recipe pull`"
+)
+@click.argument("stack_recipe_name", required=True)
+@click.option(
+    "--path",
+    "-p",
+    type=click.STRING,
+    default="zenml_stack_recipes",
+    help="Relative path at which you want to install the stack_recipe(s)",
+)
+@click.option(
+    "--yes",
+    "-y",
+    "force",
+    is_flag=True,
+    help="Force the run of the stack_recipe. This deletes the .zen folder from the "
+    "stack_recipe folder and force installs all necessary integration "
+    "requirements.",
+)
+@click.option(
+    "--shell-executable",
+    "-x",
+    type=click.Path(exists=True),
+    required=False,
+    envvar=SHELL_EXECUTABLE,
+    help="Manually specify the path to the executable that runs .sh files. "
+    "Can be helpful for compatibility with Windows or minimal linux "
+    "distros without bash.",
+)
+@pass_git_stack_recipes_handler
+@click.pass_context
+def deploy(
+    ctx: click.Context,
+    git_stack_recipes_handler: GitStackRecipesHandler,
+    stack_recipe_name: str,
+    path: str,
+    force: bool,
+    shell_executable: Optional[str],
+) -> None:
+    """Run the stack_recipe at the specified relative path.
+
+    `zenml stack_recipe pull stack_recipe_name` has to be called with the same relative
+    path before the deploy command.
+
+    Args:
+        ctx: The click context.
+        git_stack_recipes_handler: The GitStackRecipesHandler instance.
+        stack_recipe_name: The name of the stack_recipe.
+        path: The path at which you want to install the stack_recipe(s).
+        force: Force the run of the stack_recipe.
+        shell_executable: Manually specify the path to the executable that
+            runs .sh files.
+    """
+    stack_recipes_dir = Path(os.getcwd()) / path
+
+    if sys.platform == "win32":
+        logger.info(
+            "If you are running stack_recipes on Windows, make sure that you have "
+            "an associated application with executing .sh files. If you don't "
+            "have any and you see a pop-up during 'zenml stack_recipe run', we "
+            "suggest to use the Git BASH: https://gitforwindows.org/"
+        )
+
+    try:
+        _ = git_stack_recipes_handler.get_stack_recipes(stack_recipe_name)[0]
+    except KeyError as e:
+        cli_utils.error(str(e))
+    else:
+        stack_recipe_dir = stack_recipes_dir / stack_recipe_name
+        local_stack_recipe = LocalStackRecipe(
+            stack_recipe_dir, stack_recipe_name
+        )
+
+        if not local_stack_recipe.is_present():
+            ctx.invoke(
+                pull,
+                stack_recipe_name=stack_recipe_name,
+                path=path,
+                force=force,
+            )
+
+        stack_recipe_runner = (
+            [] if shell_executable is None else [shell_executable]
+        ) + [local_stack_recipe.recipes_deploy_bash_script]
+        try:
```

Comment:
I think the `track_event` should be called here instead of in the `local_stack_recipe.run_stack_recipe` method.